src/makeInfinityAdder.js

```diff
@@ -4,7 +4,31 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  const adder = function (a) {
+    let sum;
+
+    if (a === undefined) {
+      if (sum === undefined) {
+        sum = 0;
+      }
+
+      return sum;
+    }
+
+    sum = a;
+
+    return (b) => {
+      if (b === undefined) {
+        return sum;
+      }
+
+      sum += b;
+
+      return adder(sum);
```

Comment:
The recursive call to 'adder(sum)' is incorrect because it should always return a function that accepts the next number. Instead, you should return 'adder' itself without arguments.